content/en/tutorials/java-notification-app/index.md

```diff
@@ -0,0 +1,624 @@
+
+```java
+package com.example;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class MessageApplication {
+
+    public static void main(String[] args) {
+        SpringApplication.run(MessageApplication.class, args);
+    }
+
+}
+```
+
+You can now build the application using the following command:
+
+{{< command >}}
+$ mvn clean install
+{{< / command >}}
+
+If the build is successful, you will notice a `BUILD SUCCESS` message. Now that we have the application ready, let us setup the infrastructure using CloudFormation. Create a new file in ``src/main/resources` called `email-infra.yml` and add the following content:
+
+```yaml
+AWSTemplateFormatVersion: 2010-09-09
+Resources:
+  EmailQueue:
+    Type: AWS::SQS::Queue
+    Properties:
+      QueueName: email-notification-queue
+  EmailTopic:
+    Type: AWS::SNS::Topic
+    Properties:
+      TopicName: email-notifications
+
+  SnsSubscription:
+    Type: AWS::SNS::Subscription
+    Properties:
+      Protocol: sqs
+      Endpoint: !GetAtt EmailQueue.Arn
+      TopicArn: !GetAtt EmailTopic.TopicArn
+```
+
+In the above code, we have created a queue called `email-notification-queue` and a topic called `email-notifications`. We have also created a subscription between the queue and the topic, allowing any message published to the topic to be sent to the queue.
+
+## Creating the infrastructure
+
+Now that the initial coding is done, we can give it a try. Let's start LocalStack using a custom `docker-compose` setup, which includes MailHog to capture the emails sent by SES:
+
+```yaml
+version: "3.8"
+
+services:
+  localstack:
+    container_name: "${LOCALSTACK_DOCKER_NAME-localstack_main}"
+    image: localstack/localstack
+    ports:
+      - "127.0.0.1:4510-4559:4510-4559"  # external service port range
+      - "127.0.0.1:4566:4566"            # LocalStack Edge Proxy
+    environment:
+      - DEBUG=1
+      - LAMBDA_EXECUTOR=${LAMBDA_EXECUTOR-}
+      - HOST_TMP_FOLDER=${TMPDIR:-/tmp/}localstack
+      - DOCKER_HOST=unix:///var/run/docker.sock
+      - LOCALSTACK_API_KEY=${LOCALSTACK_API_KEY-}
+      - SMTP_HOST=smtp:1025
+    volumes:
+      - "${TMPDIR:-/tmp}/localstack:/tmp/localstack"
+      - "/var/run/docker.sock:/var/run/docker.sock"
+
+  smtp:
+    image: mailhog/mailhog
+    ports:
+      - "1025"
+      - "8025:8025"
+```
+
+The above `docker-compose` file will start LocalStack and pull the MailHog image to start the SMTP server (if it doesn't exist yet!) on port `8025`. You can start LocalStack using the following command:
+
+{{< command >}}
+$ LOCALSTACK_API_KEY=<your-api-key> docker-compose up -d
+{{< / command >}}
+
+Once LocalStack is started, we can deploy the CloudFormation stack (which might take a few minutes):
```

Comment:
nit: shouldn't literally take minutes, usually the deployment would be done within a couple of seconds, right?
```suggestion
Once LocalStack is started, we can deploy the CloudFormation stack (which might take a few moments):
```